### ISSUE TITLE
Canvas save breaks after a few saves if changing canvas elements. Happens in Electron 1.2.1 +, possibly in Electron 1.0 +

### ISSUE THREAD

**Issue by mgparrish** [mattdesl/electron-canvas-to-buffer#2]
Seems to be an Electron issue, was working 100% reliable in Electron  0.37.2. 

Details:
1. After a few series of changing Canvas elements and saving the canvas the file is no longer saved. When the file is not written, however the file dialog is presented, and it _appears_ that the file is saved, But it doesn't really save the file.
2. Happens saving either .jpg or .png.
3. May save OK a few times, eventually it will fail.
4. Once it fails it will not ever save canvas again until app is restarted again.
5. Tested many many canvas edits and files saves, never fails in 0.37.2 Rock Solid.
6. I am using fabric.js for canvas manipulation.  
7. The initial canvas save to file always works.
8. When the file save fails there is no error generated.
9. Using Windows 10 x64, Windows 32 bit version of Electron

I'm a green newbie so not sure how to fix or workaround, I thought maybe clearing the "buffer" after the file save since you only need the canvas rendered just immediately before writing the file. For example "nativeImage.createEmpty()", or perhaps a function could be added to index.js? But as I mentioned I'm really just learning JS and Electron. 

You can test good (made on Electron 0.37.2) vs. bad (made on Electron 1.2.1) app. Zip files are too  big to attach here. I have it on my website.  Just load a photo on the canvas and save. Both apps are the same code, just made minor changes to allow the new version 1.2.1 Electron to work, for example removing "dashes" in require statements, as you had to do in electron-canvas-to-buffer V 2.0.

Apps can be launched just click "Electron.exe" in root of the zip file.

Using Electron V 0.37.2
http://www.mgparrish.com/FabricSaveCanvasGood.zip

Using Electron V 1.21
http://www.mgparrish.com/FabricSaveCanvasBug.zip

**Comment by mgparrish** [comment]
Slight clarification on #8 above. The app does not generate an error when the write fails, but it does not generate a "Write of',filePath,'was successful' in the console during the failure. Function in "canvastest.js" in the zip files. "throw err" doesn't present itself when the file doesn't write.

function saveTheFile(){
dialog.showSaveDialog({title:'Testing a save dialog',defaultPath:'image.png'},saveCallback);
}

function saveCallback(filePath){
// as a buffer
var buffer = canvasBuffer(canvas, 'image/png')
// write canvas to file
fs.writeFile(filePath, buffer, function (err) {
 //throw err
 if(err)throw err;else console.log('Write of',filePath,'was successful');
})

**Comment by mgparrish** [comment]
I opened an issue over at the Electron repository. 6/14/2016

**Comment by mgparrish** [comment]
Update. This issue is a confirmed bug in Electron now.

https://github.com/electron/electron/issues/6050

Not exactly a nativeImage issue but a bug in Remote that effects electron-canvas-to-buffer.

Electron 0.37.2 is known good if you use electron-canvas-to-buffer, I will update here if this is resolved in future Electron builds.

**Comment by mattdesl** [comment]
Thanks for looking into this. :smile:

**Comment by mgparrish** [comment]
Confirmed it now is working correctly in Electron 1.3.1  

https://github.com/electron/electron/pull/6218 closes #6050 It appears it wasn't impliment in 1.2.6 - 1.2.8,  

1.3.0 was not tested, but 1.3.1 it works 100%

---

electron-canvas-to-buffer is a must have if you are using fabric.js and electron. Great tool

---

**Comment by mattdesl** [comment]
Hey, sorry for my lack of response, I've been traveling for a while. Is everything OK now with this module?

**Comment by mgparrish** [comment]
Yes, it's working now. The bug fix was supposed to be in Electron V 1.2.6 and it didn't work in 1.2.8 but it didn't work when I tested it. I sent a note that I had checked in 1.2.8 and it didn't work so they came back at stated it was working in 1.3.1, and I confirmed it was working 100% in that version. So I'm not clear what happened or when the fix was actually applied .... but as of 1.3.1 it is OK.
